WordPress/src/main/java/org/wordpress/android/ui/reader/repository/ReaderRepositoryCommunication.kt

```diff
@@ -2,6 +2,7 @@ package org.wordpress.android.ui.reader.repository
 
 sealed class ReaderRepositoryCommunication {
     object Success : ReaderRepositoryCommunication()
+    data class SuccessWithData<out T>(val data: T) : ReaderRepositoryCommunication()
```

Comment:
Passing interests directly to the VM via `SuccessWithData`.  We might want to keep a single `Success` state with an optional data param. Lmkwyt.

Reply:
@ashiagr - I like this idea for those non-reactive repository situations and I can go with either option. In one situation we check for the class and the other a quick optional check.

Reply:
Looks like we'll have to keep just one Success state to avoid `else` condition in `when` clause as seen [here](https://github.com/wordpress-mobile/WordPress-Android/pull/12490/commits/baab29d148ea34a32d9390cdc915c19aa48347d2#diff-8f7fbeb1ff107a11e53a17c2518cd0cfR68-R70).